src/Kestrel.Transport.Libuv/Internal/LibuvOutputConsumer.cs

```diff
@@ -48,11 +48,6 @@ public class LibuvOutputConsumer
 
                 try
                 {
-                    if (result.IsCanceled)
```

Comment:
@halter73 I ran into an issue suspicously similar to https://github.com/aspnet/KestrelHttpServer/issues/2648. My new functional tests and some of the h2spec tests were flaky because the last write on a connection would be sporadically dropped. I debugged and found that this result was returning IsCanceled:true, IsCompleted:true, and a non-empty buffer. I can't see anything that cancels it, and we want that data to get sent, so I reorganized this to prioritize sending any data given.

Reply:
@davidfowl this is what I wanted to ask you about.

Reply:
We call abort which cancels the write.

Reply:
Here https://github.com/aspnet/KestrelHttpServer/blob/58abc4b93194c82332b6cb3b83be1a5158b8f1a6/src/Kestrel.Transport.Libuv/Internal/LibuvConnection.cs#L122

Reply:
I found that one, but I couldn't find a call stack that lead to it for this scenario.

Reply:
This fixes h2spec/http2/5.4.1/2 https://github.com/aspnet/KestrelHttpServer/issues/2154#issuecomment-395503765

Reply:
Revert this. See reasoning in my SocketConnection comment.